src/payment/strategies/apple-pay/apple-pay-payment-strategy.ts

```diff
@@ -0,0 +1,166 @@
+import { assertApplePayWindow } from './is-apple-pay-window';
+
+export const validationEndpoint = 'https://bigpay.service.bcdev/api/public/v1/payments/applepay/validate_merchant';
+const appleValidationUrl = 'https://apple-pay-gateway-cert.apple.com/paymentservices/startSession';
+
+interface ApplePayPromise {
+    resolve(value: InternalCheckoutSelectors | PromiseLike<InternalCheckoutSelectors>): void;
+    reject(reason?: Error): void;
+}
+​
+export default class ApplePayPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _requestSender: RequestSender,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _sessionFactory: ApplePaySessionFactory
+    ) { }
+​
+    initialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+​
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const request = this._getBaseRequest(cart);
+        const applePaySession = this._sessionFactory.create(request);
+​
+        const { methodId } = payment;
+        const {
+            paymentMethods: { getPaymentMethodOrThrow },
+        } = await this._store.dispatch(
+            this._paymentMethodActionCreator.loadPaymentMethod(methodId, options)
+        );
+
+        const paymentMethod = getPaymentMethodOrThrow(methodId);
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder({}, options));
+​
+        assertApplePayWindow(window);
+​
+        applePaySession.begin();
+​
+        // Applepay will handle the rest of the flow so return a promise that doesn't really resolve
+        return new Promise((resolve, reject) => {
+            this._handleApplePayEvents(applePaySession, paymentMethod, { resolve, reject });
+        });
+    }
+​
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+​
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+​
+    private _getBaseRequest(cart: Cart): ApplePayJS.ApplePayPaymentRequest {
+        return {
+            countryCode: 'US',
+            currencyCode: cart.currency.code,
+            merchantCapabilities: ['supports3DS'],
+            supportedNetworks: [
+                'visa',
+                'masterCard',
+                'amex',
+                'discover',
+            ],
+            lineItems: [],
```

Comment:
Do we need to display other line items such as tax, discount etc...?

Reply:
I had to remove that since we didn't have this information as part of the cart. So feels more accurate to display the total price the shopper needs to pay.